url0e7d57473a_kyorohiro_HetimaUtil_tgz-pJ8-net_hetimatan_net_http_HttpServer3xxJ8/src/net/hetimatan/tool/SimpleHttpServer.java

```diff
@@ -0,0 +1,109 @@
+package net.hetimatan.tool;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+
+import net.hetimatan.io.file.KyoroFile;
+import net.hetimatan.io.file.KyoroFileForFiles;
+import net.hetimatan.io.filen.CashKyoroFile;
+import net.hetimatan.net.http.HttpHistory;
+import net.hetimatan.net.http.HttpServerFront;
+import net.hetimatan.net.http.HttpServerTemplate;
+import net.hetimatan.util.event.net.io.KyoroSocket;
+import net.hetimatan.util.http.HttpRequest;
+
+
+public class SimpleHttpServer extends HttpServerTemplate {
+
+	private static SimpleHttpServer server = null;
+	public static void main(String[] args) {
+		if(args.length != 1) {
+			System.out.println("java "+SimpleHttpServer.class.getSimpleName());
+			return;
+		}
+		server = new SimpleHttpServer(new File(args[0]));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code assigns a resource to a class field and there is no method that closes or returns this resource. This can result in a resource leak that slows down or crashes your system.

**Fix**
Create a method in this class that closes the following resource: *server*. After you use this resource, use this method to prevent a resource leak.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).